### DISCUSSION TITLE
Unable to pull anything from storage via createClient

### DISCUSSION THREAD

**Discussion by jorgram** [supabase/supabase#20455] [Questions] [answered]
I am using `supabase-js`. The following code returns an empy list even though I have a bucket named 'inspiring_images' with a folder and files. 
```
const { data, error } = await supabase
        .storage
        .listBuckets()
    console.log(data)
```
I have a policy that allows selects, defaults to all public roles, and has the following policy definition:

`((bucket_id = 'inspiring_images'::text) AND ((storage.foldername(name))[1] = 'images'::text))`

**Comment by GaryAustin1** [comment]
listBuckets requires policies on the buckets table not the objects table which I think you are showing.

**Reply by jorgram** [reply]
I have run:
```
create policy list_all_buckets
on storage.buckets for select using (
true
);
```
Yet this is still returning an empty list:

```
const { data, error } = await supabase
        .storage
        .listBuckets()
```

**Reply by GaryAustin1** [reply]
Are you using next.js?
If so it likes to cache URLs in many cases so could still be showing old data.

**Reply by jorgram** [reply]
Nope, just vanilla js. Using Parcel for dev server.

**Reply by GaryAustin1** [reply]
Not sure then...
![image](https://github.com/supabase/supabase/assets/54564956/f2d526ee-e4f7-441f-bbd4-3b1181e94ec5)
![image](https://github.com/supabase/supabase/assets/54564956/536dd5c4-fcde-4234-a9a1-5d0fdfed44d3)
![image](https://github.com/supabase/supabase/assets/54564956/e731508c-e71f-45e8-8ef2-78f432449c3e)
Ignore the supabases[2] part.  I have an array of supabase clients in my test instance.

**Reply by GaryAustin1** [reply]
Do you have something in the to field?
If you have authenticated then try anon and see if it works.  If so then you don't have a signed in user when you make the call.

**Reply by jorgram** [reply]
I haven't set up any kind of authentication yet I just created a bucket and set it to public and ran the above `create policy` statement so I figured it would just work (maybe that is incorrect)? 

I don't have anything in the to field in sql query. What do you mean by try anon exactly?

**Reply by GaryAustin1** [reply]
If you have nothing in the TO field of the SQL statement or the Target Roles of the UI storage bucket policies then anon is not relevant.   That was only if you had authenticated meaning you were trying to use a signed in user.  anon would test if you did not have that.  anon should work now (but so should nothing) if you are not using auth yet.

**Reply by jorgram** [reply]
I'm early in the dev process so everything in my bucket is trivial (can delete and start fresh). Are there certain steps I can take to quickly set up a test bucket with proper policies so that this code returns actual data?
```
const { data, error } = await supabase
        .storage
        .listBuckets()
```

**Reply by GaryAustin1** [reply]
I don't think there is anything in the setup process itself that would impact what you are doing and showing.  If you have what you say you have and it is going to the same instance from the client it should work.  You should also log out error, but if you are getting empty array for data versus null that points to RLS.

To do a test.
Create a bucket with the UI.
Add a policy with the UI for buckets on select with roles set to nothing (public) and true.
Run the code you show.

**Reply by jorgram** [reply]
Setting policy:
![image](https://github.com/supabase/supabase/assets/47513151/60812fca-3e72-4447-8ae7-2ce911bbb285)
![image](https://github.com/supabase/supabase/assets/47513151/5ac326d3-3f89-402d-bb0c-b948ddbb7a9d)

Simple event listener that logs buckets:
![image](https://github.com/supabase/supabase/assets/47513151/cefb7714-c06c-4221-b8b2-aa9495e2cfc5)

Output:
![image](https://github.com/supabase/supabase/assets/47513151/ae16501f-8c5c-4d94-8255-ff6d07dcf38e)

**Reply by GaryAustin1** [reply]
Are you sure your instance URL is the one you are working with?  It is rare, but I've seen that every month or so as sometimes people have multiple similar environments.

I've got no other ideas.  Your code looks good to me.  Try another instance?

**Comment by jorgram** [comment]
hmmmm here's part of my `supabase start` output:
![image](https://github.com/supabase/supabase/assets/47513151/d83a0923-a57f-492d-9254-8d384514a0d6)

That url matches `SUPABASE_URL` in my .env file.
![image](https://github.com/supabase/supabase/assets/47513151/c433be22-aa1b-4c34-9466-0d971011cebd)

Is that what you're asking?

**Reply by GaryAustin1** [reply, marked as answer]
This is a local instance and not a Supabase hosted instance?  I normally assume if someone does not mention local dev they are on the Supabase hosted instance.  I'm not that familiar with local dev and storage and what you might be able to to do to keep it from working correctly.  I would think though if your settings are not correct you would get errors versus what appears to be an RLS issue.   Returning an empty array is almost always RLS not being met.

**Reply by jorgram** [reply]
aaaahhhh I was using local dev instance variables for my `createClient` call but adding buckets on the supabase website and expecting to see the changes locally (mismatch between instance and changes I was making)

I'm correctly using my project url now in my `createClient` call instead of local dev url and things are working. That was dumb of me, my bad. Thanks for looking into this, I should've caught earlier.
